libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryClientSettings.java

```diff
@@ -68,12 +74,17 @@ private boolean isSocketFactoryUnset(SSLSocketFactory sslSocketFactory, X509Trus
     return sslSocketFactory != null && x509TrustManager != null;
   }
 
+  Builder toBuilder() {
```

Comment:
As indicated before this ends up in an infinite loop causing a `StackOverflowError` exception 💥 at runtime.

We need to implement `toBuilder` method like this one https://github.com/mapbox/mapbox-java/blob/master/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L344 (👀 the generated code by AutoValue) that would allow us to “copy” all the current values of the builder and only change the ones we’re interested in.
:eyes: https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#-initialize-a-builder-to-the-same-property-values-as-an-existing-value-instance for further information.